x-pack/legacy/plugins/apm/public/projections/transaction_groups.ts

```diff
@@ -5,39 +5,51 @@
  */
 
 import { Setup } from '../../server/lib/helpers/setup_request';
-import { getBaseTransactionGroupsProjection } from './base_transaction_groups';
 import {
-  SERVICE_NAME,
-  TRANSACTION_TYPE
+  TRANSACTION_NAME,
+  PARENT_ID
 } from '../../common/elasticsearch_fieldnames';
+import { Options } from '../../server/lib/transaction_groups/fetcher';
+import { getTransactionsProjection } from './transactions';
 import { mergeProjection } from './util/merge_projection';
 
 export function getTransactionGroupsProjection({
   setup,
-  serviceName,
-  transactionType,
-  transactionName
+  transactionName,
+  options
 }: {
   setup: Setup;
-  serviceName: string;
-  transactionType: string;
   transactionName?: string;
+  options: Options;
 }) {
-  const transactionNameFilter = transactionName
-    ? [{ term: { [TRANSACTION_TYPE]: transactionType } }]
-    : [];
+  const transactionsProjection = getTransactionsProjection({
+    setup,
+    ...(options.type === 'top_transactions'
+      ? {
+          serviceName: options.serviceName,
+          transactionType: options.transactionType
+        }
+      : {}),
+    transactionName
+  });
```

Comment:
If `transactionName` is moved into `options` you can do:
```js
  const transactionsProjection = getTransactionsProjection({
    setup,
    options,
  });
```

and then in projections/transaction.ts:
```ts
  const serviceNameFilter = options.type === 'top_transactions'
    ? [{ term: { [SERVICE_NAME]: options.serviceName } }]
    : [];
```

Then the arguments are consistent across the different methods (`setup` and `options`).